src/js/login/components/SearchHelpSignIn.jsx

```diff
@@ -15,12 +15,10 @@ class SearchHelpSignIn extends React.Component {
     const login = this.props.login;
 
     if (login.currentlyLoggedIn) {
-      if (this.props.profile.userFullName.first) {
-        const firstName = _.startCase(_.toLower(this.props.profile.userFullName.first));
-        greeting = firstName;
-      } else {
-        greeting = this.props.profile.email;
-      }
+      const firstName = _.startCase(_.toLower(
+        this.props.profile.userFullName.first || sessionStorage.userFirstName
+      ));
+      greeting = firstName || this.props.profile.email;
```

Comment:
Since `greeting` is only used in the scope of this conditional, it can just be declared here as `const greeting = ...`.